src/main/java/com/google/sps/util/DatastoreModule.java

```diff
@@ -114,7 +172,94 @@ public static String getUserProfileData(String profileId){
         userInfo.setAccess_token(userCredentials.getString("access_token"));
 
         String returnJson = new Gson().toJson(userInfo);
-       
+
         return returnJson;
     }
-}
\ No newline at end of file
+
+    /* Helper functions */
+
+    public static String stripQuotes(String input) {
+        return input.replaceAll("\"", "");
+    }
+
+    // In one line, ensures that credentials are valid
+    public static void validateUserCredentials(Entity userCredentials) throws ParseException, IOException {
+
+        if (needsToBeRefreshed(userCredentials))
+            refreshUserCredentials(userCredentials);
+
+    }
+
+    public static Boolean needsToBeRefreshed(Entity userCredentials) {
+
+        // expires_in is in seconds, get current time in seconds too
+        long currentTimeInSeconds = System.currentTimeMillis() / 1000;
+
+        // Get Creds Expiration (in seconds)
+        long expires_in = Long.parseLong(userCredentials.getString("expires_in"));
+
+        // Get time stamp
+        long credentialTimeStamp = userCredentials.getLong("timestamp");
+
+        // Determine how long (in seconds) its been since storage of access_token
+        long currentLifeSpanOfCredential = currentTimeInSeconds - credentialTimeStamp;
+
+        // If determined older than "expires_in" return true
+        return currentLifeSpanOfCredential >= expires_in;
+    }
+
+    public static void refreshUserCredentials(Entity userCredentials) throws ParseException, IOException {
+
+        // Build the refresh request
+        String refresh_request_url = buildRefreshRequest(stripQuotes(userCredentials.getString("refresh_token")));
+
+        // Executre the Refresh request
```

Comment:
typo

Reply:
Thanks for the catch, updated the comment